pkgs/top-level/all-packages.nix

```diff
@@ -2541,6 +2541,10 @@ in
 
   dgen-sdl = callPackage ../misc/emulators/dgen-sdl { };
 
+  dnschef = callPackage ../tools/networking/dnschef {
+    inherit (python3Packages) buildPythonApplication dnslib;
```

Comment:
Can we add dnslib to top-level?

Reply:
I don't understand.  dnslib is a python package already in top-level/python-packages.nix.  
    Do you prefer I not 'inherit' dnslib, or do you actually want dnslib moved to all-packages.nix?

    If the former, I could copy the style of visidata and obtain dnslib from python3Packages like this:
     dnschef = (newScope python3Packages) ../tools/networking/dnschef { };

Reply:
I don't think we should add `dnslib` to Nixpkgs top level, because it is only usable as a python library. If we forgot to package a bundled command, that's a topic for another PR.

Most python applications are packaged either as part of pythonPackages sets with an alias at top level, which I don't think is applicable here, or they are packaged here in `all-packages.nix` using `python3Packages.callPackage`. That function will also remove the need for this `inherit` and you can then remove `python3Packages` from your parameter list.

Reply:
python3Packages.callPackage it is.  Also removed python3Packages from the parameter list.